evap/static/js/datagrid.js

```diff
@@ -0,0 +1,285 @@
+class DataGrid {
+    constructor({storageKey, searchInput}) {
+        this.headers = new Map();
+        this.storageKey = storageKey;
+        this.searchInput = searchInput;
+    }
+
+    attachTo(container) {
+        this.container = container;
+        this.columns = Array.from(this.headers.keys());
+        this.rows = this.fetchRowData();
+        this.loadFromStorage();
+        this.bindEvents();
+    }
```

Comment:
The situation with `constructor` and `attachTo` is bad.
The problem I have encountered was that there is no `this` before `super`. As you can see, I exclusively set properties in the constructor and do the logic stuff in `attachTo`.

What I would have wanted in subclasses:
```javascript
constructor({filterCheckboxes, ...options}) {
    this.filterCheckboxes = filterCheckboxes;
    super(options); // would also set up row data, bind events, etc. which uses above defined properties
}
```

Adding to the confusion, the subclasses of `DataGrid` must populate `this.headers` before the call to `super.attachTo`. Note that in `attachTo`, as we have an already constructed object, there is `this`-life before `super`.

I didn't come up with alternatives, do you have?

Reply:
jakob: said he might just as well put everything into an init method.